core/src/test/scala/unit/kafka/log/LogCleanerIntegrationTest.scala

```diff
@@ -178,6 +179,40 @@ class LogCleanerIntegrationTest extends AbstractLogCleanerIntegrationTest with K
       s"but lastCleaned=$lastCleaned2", lastCleaned2 >= secondBlockCleanableSegmentOffset)
   }
 
+  @Test
+  def testTombstoneCleanWithLowThroughput() : Unit = {
+    val tombstoneRetentionMs = 1000 // this is in milliseconds -> 1 second 
+
+    val topicPartitions = Array(new TopicPartition("log-partition", 0))
+    val props = new Properties()
+    props.put(LogConfig.DeleteRetentionMsProp, "1000")
+    cleaner = makeCleaner(partitions = topicPartitions, propertyOverrides = props, backOffMs = 100L)
+
+    val log = cleaner.logs.get(topicPartitions(0))
+
+    val T0 = time.milliseconds
+    writeKeyDups(numKeys = 1, numDups = 1, log, CompressionType.NONE, timestamp = T0, 
+                 startValue = 0, step = 1, isRecordTombstone = true)
+
+    // roll the active segment
+    log.roll()
+
+    cleaner.startup()
+
+    val latestOffset: Long = log.logEndOffset
+
+    cleaner.awaitCleaned(new TopicPartition("log-partition", 0),
+                         latestOffset + 1, maxWaitMs = 5000)
+    assertEquals(log.latestDeleteHorizon, T0 + tombstoneRetentionMs)
+
+    time.sleep(tombstoneRetentionMs + 1)
+
+    // the first block should get cleaned
+    cleaner.awaitCleaned(new TopicPartition("log-partition", 0), 
+                         latestOffset + 1, maxWaitMs = 5000)
+    TestUtils.waitUntilTrue(() => log.size == 0, "Log should be empty", tombstoneRetentionMs, 100)
```

Comment:
We can be more generous with waitTimeMs. So, using the defaults for both waitTimeMs and pause is probably fine.